### PR TITLE
Add preferred Orchestrator lock

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

**Comment by hakonhall** [comment]
Given the problems with the current lock, here's the plan:
 1. Replace InterProcessSemaphoreMutex with InterProcessMutex as used in NodeRepository. This needs to be rolled out over 2 releases.
 2. Remove retry policy override tied to the current thread, and instead use the default*

This PR is the first part of (1).

*) The default retry policy is the injected curator's retry policy that uses exponential backoff.

There's a problem with exponential backoff retry policy: It looks** like Curator treats connection loss and session loss in the same way!  So the following may happen:
 a. lock aquired using ephemeral node
 b. session lost in background. Exponential retry policy creates a new session(!)
 c. someone else graphs the lock
 d. operation to write to ZK is done under the false assumption of the lock being held

**)  I'm looking at RetryLoop::shouldRetry, but other parts of the code could do it differently - I'm no expert on Curator. The Curator documentation is non-existent.

**Review by bjorncs — APPROVED** [review]
LGTM

File: orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java [hunk 1]
```diff
@@ -86,7 +80,7 @@ public ApplicationInstanceStatus getApplicationInstanceStatus() {
      */
     @Override
     public MutableStatusRegistry lockApplicationInstance_forCurrentThreadOnly(ApplicationInstanceReference applicationInstanceReference) {
-        return lockApplicationInstance_forCurrentThreadOnly(applicationInstanceReference, 10, TimeUnit.SECONDS);
```
[the remark this answers is not in the document]
Comment (hakonhall) [reply orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java]: This is part of the code to be removed - the new code uses Duration.

**Comment by hakonhall** [comment]
Vespa 6.143.214

File: orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (bjorncs) [reply orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java]: :+1:

**Pull request merged**